docs/en/development/data_structures.md

```diff
@@ -5,4 +5,879 @@ slug: /internals
+
+Records the parent directory of some files. The parent directory of most files is recorded in the Parent field of the attribute; however, for files that have been created with hard links, there may be more than one parent directory, so the Parent field is set to 0, and all parent inodes are recorded independently, as follows
+
+```
+inode -> parentInode, links
+```
+
+where links is the count of the parentInode, because multiple hard links can be created in the same directory, and these hard links share the inode.
+
+#### 3.1.8 Chunk
+
+Records information about each Chunk, as follows
+
+```
+inode, index -> []Slices
+```
+
+where inode is the inode number of the file to which the Chunk belongs, and index is the number of all Chunks in the file, starting from 0. The Chunk value is an array of Slices. Each Slice represents a piece of data written by the client, and is appended to this array in the order of writing time. When there is an overlap between different Slices, the later Slice is used.
+
+```go
+type Slice struct {
+	Pos  uint32 // offset of the Slice in the Chunk
+	ID   uint64 // ID of the Slice, globally unique
+	Size uint32 // size of the Slice
+	Off  uint32 // offset of valid data in this Slice
+	Len  uint32 // size of valid data in this Slice
+}
+```
+
+This structure is encoded and saved in binary format, taking up 24 bytes.
+
+#### 3.1.9 SliceRef
+
+Records the reference count of a Slice, as follows
+
+```
+sliceId, size -> refs
+```
+
+Since the reference count of most Slice is 1, to reduce the number of related entries in the database, the actual value minus 1 is used as the stored count value in Redis and TKV. In this way, most of the Slices have a refs value of 0, and there is no need to create related entries in the database.
+
+#### 3.1.10 Symlink
+
+Records the location of the softlink file, as follows
+
+```
+inode -> target
+```
+
+#### 3.1.11 Xattr
+
+Records extended attributes (Key-Value pairs) of a file, as follows
+
+```
+inode, key -> value
+```
+
+#### 3.1.12 Flock
+
+Records BSD locks (flock) of a file, specifically.
+
+```
+inode, sid, owner -> ltype
+```
+
+where sid is the client session ID, owner is a string of numbers, usually associated with a process, and ltype is the lock type, which can be 'R' or 'W'.
+
+#### 3.1.13 Plock
+
+Record POSIX record locks (fcntl) of a file, specifically
+
+```
+inode, sid, owner -> []plockRecord
+```
+
+Here plock is a more fine-grained lock that can lock only a certain segment of the file.
+
+```go
+type plockRecord struct {
+	ltype uint32 // lock type
+	pid   uint32 // process ID
+	start uint64 // start position of the lock
+	end   uint64 // end position of the lock
+}
+```
+
+This structure is encoded and stored in binary format, taking up 24 bytes.
+
+#### 3.1.14 DelFiles
+
+Records the list of files to be cleaned. Since data cleanup of files is an asynchronous and potentially time-consuming operation that can be interrupted by other factors, it is tracked by this list.
```

Comment:
```suggestion
Records the list of files to be cleaned. It may be needed as data cleanup of files is an asynchronous and potentially time-consuming operation that can be interrupted by other factors.
```